src/systems/filecoin_markets/deal/deal.id

```diff
@@ -3,31 +3,43 @@ import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
 import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
 import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
 
 type DealID UVarint
 type DealCID ipld.CID
+type ProposalCID ipld.CID
 type PayloadCID ipld.CID
 type Signature struct {}  // TODO
 
 type StorageDealProposal struct {
-    // The data hashed in a form that is compatible with the proofs system.
-    // Reference for actual storage in a sector.
-    PieceCID                piece.PieceCID  // 35 bytes CommP
-    PieceSize               piece.PieceSize
-    Client                  addr.Address
-    Provider                addr.Address
-    ProposalExpiration      block.ChainEpoch
-    DealExpiration          block.ChainEpoch  // potentially change to DealDuration
-    StoragePrice            actor.TokenAmount  // potentially per epoch
-    ProviderDealCollateral  actor.TokenAmount
-    ClientDealCollateral    actor.TokenAmount  // potentially collapse into one with ProviderDealCollateral
-    ProposerSignature       Signature
+    PieceCID                      piece.PieceCID  // 35 bytes CommP
+    PieceSize                     piece.PieceSize
+    Client                        addr.Address
+    Provider                      addr.Address
+    ClientSignature               Signature
+
+    StartEpoch                    block.ChainEpoch  // a deal is invalid if it is published to the chain after StartEpoch
+    EndEpoch                      block.ChainEpoch
+    StoragePricePerEpoch          actor.TokenAmount
+    ProviderCollateralPerEpoch    actor.TokenAmount
+    ClientCollateralPerEpoch      actor.TokenAmount  // potentially collapse into one with ProviderDealCollateral
```

Comment:
Yes, will do. The reason why we have `ClientCollateralPerEpoch` (which needs be more than a `MIN_CLIENT_DEAL_COLLATERAL_PER_DEAL`) is to incur a cost for self dealing. We are still figuring the economic construction here and its relationship with other constructions, but adding it here as a placeholder. It might collapse into `ProviderCollateralPerEpoch` in the future or be removed.

Reply:
The reason that it is `PerEpoch` instead of just `ClientCollateralPerEpoch` (and similarly for `ProviderCollateral`) is so that we have the ability to disincentivize or penalize self-dealed storage deals that last a long time with the minimum price and collateral.

Reply:
All collateral will only be unlocked when the storage deal is no longer counting towards power (in the current spec, that is when the whole sector expires). I expect changes there but these structs set up the right placeholders and building blocks for whichever decisions we make later.